src/chaperone.rs

```diff
@@ -0,0 +1,186 @@
+    Error,
+    BaseStationUninitialized,
+    BaseStationConflict,
+    PlayAreaInvalid,
+    CollisionBoundsInvalid,
+    Unknown(u32),
+}
+
+impl From<sys::ChaperoneCalibrationState> for ChaperoneCalibrationErrorState {
+    fn from(state: sys::ChaperoneCalibrationState) -> Self {
+        use self::ChaperoneCalibrationErrorState::*;
+        match state {
+            sys::ChaperoneCalibrationState_Error => Error,
+            sys::ChaperoneCalibrationState_Error_BaseStationUninitialized => {
+                BaseStationUninitialized
+            }
+            sys::ChaperoneCalibrationState_Error_BaseStationConflict => BaseStationConflict,
+            sys::ChaperoneCalibrationState_Error_PlayAreaInvalid => CollisionBoundsInvalid,
+            _ => Unknown(state as u32),
+        }
+    }
+}
+
+/// State of Chaperone calibration.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub enum ChaperoneCalibrationState {
+    Ok,
+    Warning(ChaperoneCalibrationWarningState),
+    Error(ChaperoneCalibrationErrorState),
+    Unknown(u32),
+}
+
+impl From<sys::ChaperoneCalibrationState> for ChaperoneCalibrationState {
+    fn from(state: sys::ChaperoneCalibrationState) -> Self {
+        use self::ChaperoneCalibrationState::*;
+        match state {
+            1 => Ok,
+            100...199 => Warning(ChaperoneCalibrationWarningState::from(state)),
+            200...299 => Error(ChaperoneCalibrationErrorState::from(state)),
+            _ => Unknown(state as u32),
+        }
+    }
+}
+
+impl Chaperone {
+    /// Get the current state of Chaperone calibration.
+    /// This state can change at any time during a session due to physical base station changes.
+    /// (NOTE: Some of these error codes are never returned as implementation for the error states
+    /// is still a work in progress.)
+    pub fn get_calibration_state(&self) -> ChaperoneCalibrationState {
+        unsafe { self.0.GetCalibrationState.unwrap()() }.into()
+    }
+
+    /// Returns the width and depth of the Play Area.
+    pub fn get_play_area_size(&self) -> Option<(f32, f32)> {
+        let mut x: f32 = 0.0;
+        let mut z: f32 = 0.0;
+        let is_ok = unsafe { self.0.GetPlayAreaSize.unwrap()(&mut x, &mut z) };
+        if is_ok {
+            Some((x, z))
+        } else {
+            None
+        }
+    }
+
+    /// Returns the 4 corner positions of the PlayArea.
+    pub fn get_play_area_rect(&self) -> Option<HmdQuad_t> {
+        let mut play_area_rect = HmdQuad_t {
+            vCorners: [HmdVector3_t { v: [0.0; 3] }; 4],
+        };
+        let is_ok = unsafe { self.0.GetPlayAreaRect.unwrap()(&mut play_area_rect) };
+        if is_ok {
+            Some(play_area_rect)
+        } else {
+            None
+        }
+    }
+
+    // Do not understand what this does
+    /// Reload chaperone info
+    fn reload_info(&mut self) {
+        unsafe { self.0.ReloadInfo.unwrap()() };
+    }
+
+    // Do not understand what this does
+    /// Set scene color
```

Comment:
The color of the chaperone wireframe, perhaps?

Reply:
That did not work with my Vive; I tried the reload afterward but still no dice.